tools/NewRelicHelper.psm1

```diff
@@ -176,101 +184,109 @@ function update_azure_service_definition([System.__ComObject] $project){
-        $licenseKeyRoleInstanceValueNode = $xml.CreateElement('RoleInstanceValue','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
-        $licenseKeyRoleInstanceValueNode.SetAttribute('xpath','/RoleEnvironment/CurrentInstance/ConfigurationSettings/ConfigurationSetting[@name=''NewRelic.LicenseKey'']/@value')
-        $licenseKeyVariableNode.SetAttribute('name','LICENSE_KEY')
-        $licenseKeyVariableNode.AppendChild($licenseKeyRoleInstanceValueNode)
-        $environmentNode.AppendChild($licenseKeyVariableNode)
-        
-        $taskNode.AppendChild($environmentNode)
-        $startupNode.AppendChild($taskNode)
-        
-        $modifiedStartUp = $modified.StartUp
-        if($modifiedStartUp -eq $null){
-        	$modified.PrependChild($startupNode)
-        }
-        else{
-        	$nodeExists = $false
-        	foreach ($i in $modifiedStartUp.Task){
-        		if ($i.commandLine -eq "newrelic.cmd"){
-        			$nodeExists = $true
-        		}
-        	}
-        	if($NewRelicTask -eq $null -and !$nodeExists){
-        		$modifiedStartUp.AppendChild($taskNode)
-        	}
-        }
-        
-        if($modified -ne $null -and $isWorkerRole -eq 'true'){
-        
-        	#Generate the environment variables for worker role instrumentation
-        	$runtimeNode = $xml.CreateElement('Runtime','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
-        	$runtimeEnvironmentNode = $xml.CreateElement('Environment','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
-        	
-        	#Enables profiling for all CLR based applications
-        	$variableCEPNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
-        	$variableCEPNode.SetAttribute('name','COR_ENABLE_PROFILING')
-        	$variableCEPNode.SetAttribute('value','1')
-        	
-        	#Profiler guid associated with the New Relic .NET profiler
-        	$variableCPNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
-        	$variableCPNode.SetAttribute('name','COR_PROFILER')
-        	$variableCPNode.SetAttribute('value','{71DA0A04-7777-4EC6-9643-7D28B46A8A41}')
-        	
-        	#Helps Azure Workers find the newrelic.config
-        	$variableNHNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
-        	$variableNHNode.SetAttribute('name','NEWRELIC_HOME')
-            $variableNHNode.SetAttribute('value','D:\ProgramData\New Relic\.NET Agent\')
+        #Make sure a matching app was found
+        if($modified -ne $null){
+            #Generate the variable for the startup task that will be used to check to see if we need to issue a restart on the W3SVC 
+            $variableIsWorkerRoleNode = $xml.CreateElement('Variable','http://schemas.microsoft.com/ServiceHosting/2008/10/ServiceDefinition')
```

Comment:
Absolutely.  I was trying to avoid cascading refactors for this PR.  Another will come with the suggested changes.